app/models/audio_version_template.rb

```diff
@@ -43,42 +43,58 @@ def touch_audio_versions
   end
 
   def validate_audio_version(audio_version)
-    [audio_file_count_errors(audio_version),
-     audio_length_errors(audio_version)].join(' ').strip
+    audio_file_count_errors(audio_version) || audio_length_errors(audio_version)
+  end
+
+  def validate_audio_file(audio_file)
+    audio_file_content_type_errors(audio_file)
   end
 
   private
 
   def audio_file_count_errors(audio_version)
-    file_count_errors = ''
     num_audio_files = audio_version.audio_files.count
     if !segment_count.nil? && audio_version.audio_files.count != segment_count
-      file_count_errors << "Audio version #{audio_version.label} has #{num_audio_files} " +
-                           "audio files but must have #{segment_count} segments."
+      "Audio version #{audio_version.label} has #{num_audio_files} audio " +
+        "files but must have #{segment_count} segments."
     end
-    file_count_errors
   end
 
   def audio_length_errors(audio_version)
-    length_errors = ''
-    shorter_than_min = audio_version.length < length_minimum
+    shorter_than_min = length_minimum != 0 && audio_version.length < length_minimum
     longer_than_max = length_maximum != 0 && audio_version.length > length_maximum
     if shorter_than_min || longer_than_max
-      length_errors << "Audio version '#{audio_version.label}' is " +
-                       "#{audio_version.length.to_time_string} long but must be "
+      length_errors = "Audio version '#{audio_version.label}' is " +
+                      "#{audio_version.length.to_time_string} long but must be "
       length_errors << if length_maximum != 0
                          "#{length_minimum.to_time_string} - " +
                          "#{length_maximum.to_time_string} long."
                        else
                          "more than #{length_minimum.to_time_string} long."
                        end
+      length_errors
     end
-    length_errors
   end
 
   def max_is_greater_than_min_if_set
     if length_maximum != 0 && length_minimum > length_maximum
       errors.add(:length_mininum, 'must be less than or equal to length maximum')
     end
   end
+
+  def audio_file_content_type_errors(audio_file)
+    if content_type.present?
+      tpl_type = content_type.split('/').first
+      file_type = audio_file.content_type.try(:split, '/').try(:first)
+      if content_type == 'audio/mpeg' && audio_file.content_type != 'audio/mpeg'
+        "Audio file '#{audio_file.label}' is not an mp3" # very specific on these
+      elsif tpl_type != file_type
+        "Audio file '#{audio_file.label}' is not #{articleize(tpl_type)}"
+      end
+    end
+  end
+
+  def articleize(word)
```

Comment:
neat, but is this necessary? could we re-word to  "Audio file '#{audio_file.label}' is not in #{tpl_type} format"

and actually, maybe we should stop saying "Audio file" since videos and other files may come through here - maybe just "File is not in mp3 format"?

Reply:
Hmm... yeah, that "Audio File" thing is annoying, since you see it in the frontend a lot.  Will update.